src/Backend/CompilerImpl/DebianCompiler.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as cp from "child_process";
+import * as vscode from "vscode";
+
+import { pipedSpawnSync } from "../../Utils/PipedSpawnSync";
+import { Command } from "../Command";
+import { Compiler } from "../Compiler";
+import { PackageInfo, ToolchainInfo, Toolchains } from "../Toolchain";
+import {
+  DebianArch,
+  DebianRepo,
+  DebianToolchain,
+} from "../ToolchainImpl/DebianToolchain";
+import { Version } from "../Version";
+
+export class DebianCompiler implements Compiler {
```

Comment:
The name `DebianCompiler` is a little weird. I will update the name to the appropriate name later.